Source/JavaScriptCore/b3/B3LowerToAir.cpp

```diff
@@ -2875,7 +2875,7 @@ class LowerToAir {
                 if (incrementArg) {
                     append(relaxedMoveForType(address->type()), tmp(address->child(0)), tmp(address));
                     append(opcode, incrementArg, tmp(memory));
-                    m_locked.add(address);
+                    commitInternal(address);
```

Comment:
Can you explain why this is necessary here? It looks like `address` is now computed and stored to `tmp`, so it is not internal, right?

Reply:
Since `MoveWithIncrement` increments `address` by `offset` internally, I think we should `commitInternal(address)` in case other pattern matching optimization related to `address` might happen. If so, we might compute `Add(address, offset)` internally twice.

Reply:
I will add that as a comment to code.

Reply:
Updated. Thanks! That's a really good question!